source/common/quic/client_connection_factory_impl.cc

```diff
@@ -43,11 +43,11 @@ std::shared_ptr<quic::QuicCryptoClientConfig> PersistentQuicInfoImpl::cryptoConf
 
 PersistentQuicInfoImpl::PersistentQuicInfoImpl(
     Event::Dispatcher& dispatcher, Network::TransportSocketFactory& transport_socket_factory,
-    TimeSource& time_source, Network::Address::InstanceConstSharedPtr server_addr,
-    const quic::QuicConfig& quic_config, uint32_t buffer_limit)
+    TimeSource& time_source, uint32_t remote_port, const quic::QuicConfig& quic_config,
+    uint32_t buffer_limit)
     : conn_helper_(dispatcher), alarm_factory_(dispatcher, *conn_helper_.GetClock()),
       server_id_{getConfig(transport_socket_factory).serverNameIndication(),
-                 static_cast<uint16_t>(server_addr->ip()->port()), false},
+                 static_cast<uint16_t>(remote_port)},
```

Comment:
since a port can only be a 16 bit value, would it make sense to have |remote_port| be a uint16_t, or would that just push the problem up the stack because uint32_t is the idiomatic way to talk about a port in Envoy?

Reply:
yeah I think it's going to cause problems somewhere, unless we do a big migration